app/get_proc_arg.go

```diff
@@ -0,0 +1,222 @@
+	dir, err := discoverProcArg(kubeClient, managerImage, DetectFlexvolumeName, detectFlexVolumeScript)
+	if err != nil {
+		errors.Wrap(err, "Failed to get arg volume-plugin-dir in proc kubelet")
+		return "", err
+	}
+	if dir == "" {
+		dir = DefaultFlexvolumeDir
+	}
+	return dir, nil
+}
+
+func discoverKubeletRootDir(kubeClient *clientset.Clientset, managerImage string) (string, error) {
+	// try to detect root-dir in proc kubelet
+	detectKubeletRootDirScript := fmt.Sprintf(DetectKubeletArgScript, ArgKubeletRootDir, DefaultRootDir)
+	rootDir, err := discoverProcArg(kubeClient, managerImage, DetectRootDirName, detectKubeletRootDirScript)
+	if err != nil {
+		errors.Wrap(err, "Failed to get arg root-dir in proc kubelet")
+		return "", err
+	}
+	// not found in proc kubelet. then try to detect root-dir in proc k3s
+	if rootDir == "" {
+		rootDir, err = discoverProcArg(kubeClient, managerImage, DetectRootDirName, DetectK3SDataDirScript)
+		// proc k3s exists.
+		if rootDir != "" {
+			rootDir = filepath.Join(rootDir, "/agent/kubelet")
+		}
+	}
+	if err != nil {
+		errors.Wrap(err, "Failed to get arg root-dir in proc k3s")
+		return "", err
+	}
+	// not found. use default
+	if rootDir == "" {
+		rootDir = DefaultRootDir
+	}
+	return rootDir, nil
+}
+
+func discoverProcArg(kubeClient *clientset.Clientset, managerImage, name, script string) (string, error) {
+	namespace := os.Getenv(types.EnvPodNamespace)
+	if namespace == "" {
+		return "", fmt.Errorf("Cannot detect pod namespace, environment variable %v is missing", types.EnvPodNamespace)
+	}
+
+	if err := deployDetectionPod(kubeClient, namespace, managerImage, name, script); err != nil {
+		logrus.Errorf("Failed to deploy kubelet arg detection pod %v: %v", name, err)
+		return "", err
+	}
+
+	defer func() {
+		if err := kubeClient.CoreV1().Pods(namespace).Delete(name, &metav1.DeleteOptions{}); err != nil {
+			logrus.Warnf("Failed to delete kubelet arg detection pod %v: %v", name, err)
+		}
+	}()
+
+	completed := false
+	for i := 0; i < DetectPodMaxPolls; i++ {
+		if pod, err := kubeClient.CoreV1().Pods(namespace).Get(name, metav1.GetOptions{}); err != nil {
+			logrus.Warnf("Failed to get kubelet arg detection pod %v: %v", name, err)
+		} else if pod.Status.Phase == v1.PodSucceeded {
+			completed = true
+			break
+		} else {
+			logrus.Debugf("kubelet arg detection pod %v in phase: %v", name, pod.Status.Phase)
+		}
+		time.Sleep(1 * time.Second)
+	}
+	if !completed {
+		return "", fmt.Errorf("kubelet arg detection pod %v didn't complete within %d seconds", name, DetectPodMaxPolls)
+	}
+
+	kubeletArg, err := getPodLogAsString(kubeClient, namespace, name)
+	if err != nil {
+		return "", err
+	}
+	return kubeletArg, nil
+}
+
+func deployDetectionPod(kubeClient *clientset.Clientset, namespace, managerImage, name, script string) (err error) {
+	privileged := true
+	_, err = kubeClient.CoreV1().Pods(namespace).Create(&v1.Pod{
```

Comment:
Since you're not doing anything with err, just use `_, err := xxx` and remove the `err` variable in the function signature.

We don't prefer naked return normally. It makes code harder to read (and sometimes easier to get it wrong).